Library/Homebrew/livecheck.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class Livecheck
+  attr_reader :skip_msg
+
+  def initialize
+    @regex = nil
+    @skip = false
+    @skip_msg = nil
+    @url = nil
+  end
+
+  def regex(pattern = nil)
+    return @regex if pattern.nil?
+
+    @regex = pattern
+  end
+
+  def skip(skip_msg = nil)
+    @skip = true
+    @skip_msg = skip_msg.blank? ? nil : skip_msg
+  end
+
+  def skip?
+    @skip
+  end
+
+  def url(val = nil)
+    return @url if val.nil?
+
+    @url = val
+  end
+
+  def to_hash
+    Hash[instance_variables.map { |name| [name, instance_variable_get(name)] }]
```

Comment:
I think it would be better to manually specify this hash; it's easier to read and figure out what the format is.